### PR TITLE
[CS-3871]: Add SecureStore package and initial models

### PR DESCRIPTION
**Pull request by danibonilha**
### Description

This PR adds the SecureStore package with Pin and seedPhrase models.

- Configured jest to work with expo
- Setup new env var to secure keys, remember to sync app_vars 
- Setup test env to be able to mock implementation 
- Added test to Pin model
- Added initial auth model (lot's of todo still, will add test later)

PS: Didn't add tests to the seedPhrase piece, bc I'm having some issues mocking the encryptor, will do on following up PRs

### CODE REVIEW

**Review by paulinhapenedo — APPROVED** [review]
Looks good 👏

File: cardstack/src/models/secure-storage.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    logger.sentry(`No value for key: ${secureKey}`, e);
+
+    return null;
+  }
+};
+
+// PIN
+const savePin = (pin: string) => SecureStore.setItemAsync(keys.AUTH_PIN, pin);
+
+const getPin = () => getSecureValue(keys.AUTH_PIN);
```
Comment (oidoug) [inline cardstack/src/models/secure-storage.ts:36]: Are we going to keep the unlock pin in a global state? I'm assuming we'll need that or we'll need the user to unlock the pin every time we need it.

**Review by oidoug — APPROVED** [review]
Pretty good! A lot cleaner than our current model.
Just left a comment / question for you.

File: cardstack/src/models/secure-storage.ts (hunk 1 above)
Comment (danibonilha) [reply cardstack/src/models/secure-storage.ts:36]: We'll need in two flows, but we can talk on DM about it

**Pull request merged**